### ISSUE TITLE
Coinbase defaults to '0000000..0'

### ISSUE THREAD

**Issue by androlo** [ethereum/cpp-ethereum#1435]
The address '00000000...0' is receiving the payouts for me on latest dev.

**Comment by androlo** [comment]
Doesn't seem like it can be changed. Tried setting coinbase to the current address through the cli using 'setaddress' but did not change. balance did not go up from mining. also segfaulted as i exited from interactive. This used to work fine a few days ago btw (not a crappy build or something).

```
Press Enter
> balance
Current balance: 0 wei = 0 wei
*** [ 05:21:34 | miner-2 ] Completed d7dfdfb7… 961ecac6… 131136 1
--- [ 05:21:34 | miner-2 ] Completing mine!
*** [ 05:21:34 | miner-2 ] Mined f8fe51fb… (parent: 7745b220… )
=B= [ 05:21:35 | eth ] Attempting import of f8fe51fb… ...
=B= [ 05:21:35 | eth ]    Imported and best 786496 . Has 0 siblings. Route: [ f8fe51fb…, ]
*** [ 05:21:35 | eth ] New block on chain: Restarting mining operation.
*** [ 05:21:35 | eth ] noteChanged( { 0000000000000000000000000000000000000000000000000000000000000000, 0000000000000000000000000000000000000000000000000000000000000001 } )
-N- [ 05:21:35 | ethsync ] Sending a new block (current is f8fe51fbca0c9fb0227fa65be50ce0562520cab9e487f6f5ec133ce798ead1f0 , was 7745b220509ecd6989784c94172f775f86d2845177cb1fb5e36ff15b6fa1d93c )
Press Enter   
> balance
Current balance: 0 wei = 0 wei
Press Enter
> address
Current address:
c040712e77cc04f4b01fcb81e0b1cfb9ba2a8e82
Press Enter
> balance
Current balance: 0 wei = 0 wei
Press Enter
> exit
Segmentation fault (core dumped)
```

**Comment by caktux** [comment]
Confirmed, same issue with payouts going to 0x000..., just no segfault.

**Comment by ethers** [comment]
ah this explains why the beneficiary of all the recent blocks are 00000000...0 and why i wasn't getting wei either

**Comment by androlo** [comment]
Thinking maybe it's a bug, because I'm starting it using `./eth -a 24d59a00f0c0da311b51ad36da0c6844478290b3`, which is the address i get from typing 'address' in interactive. I see this in the log.

```
Transaction Signer: 24d59a00f0c0da311b51ad36da0c6844478290b3
Mining Benefactor: 24d59a00f0c0da311b51ad36da0c6844478290b3
```

Still payouts are sent to the 0 address.

I normally also don't get the segfault, that was a one time thing. Not sure why, have not been able to reproduce.

**Comment by ethers** [comment]
i think it's a bug too.  with eth, you will probably want to use both -a and -s switches (since it will only be confusing to have a different benefactor and signer, unless you really want them to be different)

**Comment by youlian** [comment]
probably related to my segfault issue from some days ago... cause i am getting zeroes as well... doing some digging myself after my issue was labeled "unconfirmed"...

**Comment by gavofyork** [comment]
fixed by 995aae9..35960c8
